app/models/alchemy/page_extension.rb

```diff
@@ -1,8 +1,23 @@
 # frozen_string_literal: true
 
 Alchemy::Page.module_exec do
+  include Alchemy::Page::Annotations
+
   after_commit :flush_cache
+
   def flush_cache
     Europeana::PageCacheJob.perform_later id
   end
+
+  def exhibition
+    @exhibition ||= depth == 2 ? self : self_and_ancestors.detect { |page| page.depth == 2 }
+  end
+
+  def credits
```

Comment:
Are there not existing AR associations between these models that could be reused instead of these custom joins?

Reply:
Not efficiently and without defining all the table names for the polymorphic joins like 'content'/'essence' which can be various types, at least I couldn't get it working.